rdmalib/include/rdmalib/buffer.hpp

```diff
@@ -2,192 +2,396 @@
 #ifndef __RDMALIB_BUFFER_HPP__
 #define __RDMALIB_BUFFER_HPP__
 
-#include <cstdint>
-#include <utility>
-
-#include <cereal/cereal.hpp>
+#include <sys/mman.h>
 
-#ifdef USE_LIBFABRIC
+#include <rdma/fabric.h>
```

Comment:
Shouldn't this include be behind `#ifdef`? What if we compile without libfabric but with ibverbs?

Reply:
Since the code contains symbols from this header, I was getting compiler issues when it was not included. I discussed this issue with Marcin earlier in the summer and he said that this was okay for now.

Reply:
@mattnappo The problem is not inclusion - the problem is that the include should be guarded by `#ifdef`. If I compile just with verbs but not libfabric, then will it work?